concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -223,8 +223,9 @@ readBlobBSFromHandle BlobStoreAccess{..} (BlobRef offset) = mask $ \restore -> d
             hSeek bhHandle AbsoluteSeek (fromIntegral offset)
             esize <- decode <$> BS.hGet bhHandle 8
             case esize :: Either String Word64 of
-                Left e -> error e
-                Right size -> BS.hGet bhHandle (fromIntegral size)
+                Right size | offset + size <= fromIntegral bhSize ->
+                             BS.hGet bhHandle (fromIntegral size)
+                _ ->  throwIO $ userError "Attempted to read beyond the blob store end"
```

Comment:
```suggestion
                Right size | offset + size <= fromIntegral bhSize ->
                             BS.hGet bhHandle (fromIntegral size)
                             | otherwise -> throwIO $ userError "Attempted to read beyond the blob store end"
```

Reply:
This isn't isn't going to be the same condition: the case where the blob size read failed (`Left _`) won't be covered.